classes/module/Module.php

```diff
@@ -234,6 +234,11 @@ abstract class ModuleCore implements ModuleInterface
 
     public static $hosted_modules_blacklist = ['autoupgrade'];
 
+    public static function resetStaticCache()
+    {
+        static::$_INSTANCE = [];
+    }
+
```

Comment:
I have a doubt about this function 🤔 If we use it for something else than the tests, the only reason I see why we would use it is to get a more recent version of a module (after an upgrade for instance) but as we use `include_once` on the module, we won't get the new version.

Reply:
Except in test environment I don't think this is useful indeed You're right that the `include_once` prevents loading a new class in the same process Maybe this will be needed someday But this is still useful for tests especially since all the module classes have a copied reference of the context

But so far only one test class uses this and as you said it's isolated so I'm gonna check if this is still useful, if not I'll clean it